Add devise - Delivers #132282915 and #133202531

aganpule opened the pull request:
Added devise to our app so we we have sign up/sign in functionality. Added Sign Up and Sign In links to the nav bar. Added cucumber tests for authentication.

jeremyjpark committed:
ag/jp/ss - Add cukes for user auth

jeremyjpark committed:
ag/jp/ss - Add Devise user model to Datong

aganpule committed:
ag- Add more scenarios

aganpule committed:
ag/jp- Add Sign Up/Sign In links to nav bar

aganpule committed:
ls/ag - Add more scenarios and step definitions

susannasouv commented:
Might want to pull from master, get back on this branch, do `git merge master`, and push

susannasouv commented on features/step_definitions/static_page_steps.rb:
```diff
@@ -20,3 +20,24 @@ def with_scope(locator)
+end
+
+Given /^I do not have an account$/ do
+  pass
```
should this be pass or pending? I don't really know what's the difference

jeremyjpark committed:
jp/ss - Add additional tests for logging out

Also changed a few of the existing tests to take advantage of existing step
definitions and added pending step defs for the steps that required them. In
addition, I combined the tests from the story about logging out into this
branch.

jeremyjpark committed:
jp - Add test for resources tab

jeremyjpark committed:
jp - Add tests for events tab

jeremyjpark committed:
jp - add tests for contact us html

jeremyjpark committed:
Merge pull request #19 from team-datong/AddDeviseTest

Add devise test

jeremyjpark committed:
Merge pull request #20 from team-datong/AddTestsForContactUs

Add tests for contact us html

jeremyjpark committed:
Merge pull request #21 from team-datong/CreateResourcesTabTest

Add test for resources tab

jeremyjpark committed:
Merge pull request #22 from team-datong/CreateEventsTabTest

Add tests for events tab

jeremyjpark committed:
jp/ss - Add tests, features, and step definitions to repo

Also added a database cleaner and a cli debugger to the Gemfile as we were
finding that we had a few bugs that showed up in test. Also changed the
codeclimate.yml file to exclude some autogenerated devise views.

jeremyjpark committed:
jp/ss - Made test cases for story pass.

Users are now able to sign up, sign in, and sign out. There is now also
functionality given for free by devise to prevent users from logging in if they
have an incorrect email or login.

jeremyjpark commented:
We need to properly code review this pull request before merging.
